### PR TITLE
Games available scope

### PR DESCRIPTION
**Pull request by Phatkid98**
-added games.available scope method to game.rb
-able to loop through "available" games on index.html.erb

### CODE REVIEW

**Comment by fistoriza** [comment]
hi Miguel. It seems that when executing .available method, there is a 'undefined local variable or method 'user' error. Did this work when you tested it?

**Comment by fistoriza** [comment]
I think the user.count should be just equal 1, instead of < 2 because a game will only be available if only there is one player.

**Comment by Phatkid98** [comment]
Hey fisto, I think i actually need to change "user" to "user_id". But if I change the user.count to eq(1) then that means that a game with no players in it will NOT be available. So as long as the there aren't two players, it should be available. Know what I mean?

**Comment by fistoriza** [comment]
hi @Phatkid98 i think there will be no game with no player because each game should be created by a user. all game will at least have 1 player and maximum 2 player. if it is only 1 player on the game, then it's available. if another user joins the game, the game will be unavailable. please cmiiw.

**Comment by Phatkid98** [comment]
@fistoriza , I see what your are saying. we could definitely change that, I don't think it would affect anything else. I'll work on that and a test and commit it up

**Comment by kboygit** [comment]
hey did anyone review and give thumbs up?

File: app/views/games/index.html.erb [hunk 1]
```diff
@@ -60,3 +60,8 @@
 </div>
 
 
+<% @games.available.each do |game| %>
+  <%= game.inspect %>
```
Comment (ilyakrasnov) [inline app/views/games/index.html.erb]: should be probably sth else than game.inspect.. maybe a link to game show page?

**Comment by fistoriza** [comment]
Hi @ilyakrasnov I tried to run the games.available in console, but did not produce the desired outcome. do you think the method is all good?

**Comment by kirbygit** [comment]
@fistoriza try Game.all

**Comment by fistoriza** [comment]
@kirbygit Game.all works...but Game.all.available no.

**Comment by kirbygit** [comment]
@fistoriza where is the function available located?

**Comment by fistoriza** [comment]
@kirbygit in game.rb model.

**Comment by fistoriza** [comment]
@kirbygit i tried to modify the method to be like this: 
`	scope :available,	-> { where(private: false).where(:game_users == 2) }`

it works, but then i need to change the association between user, game and piece a bit...

**Comment by kirbygit** [comment]
@fistoriza what did you change it says undefined method count

**Comment by kirbygit** [comment]
@Phatkid98 hey you should check if that works first before merging,

**Comment by kirbygit** [comment]
[Games.available scope](https://trello.com/c/Gj7NzcSC/14-games-available-scope)

**Comment by ilyakrasnov** [comment]
Did you try Gama.available?

--
Ilya Krasnov
LinkedIn <https://www.linkedin.com/in/ikrasnov>

**Comment by Phatkid98** [comment]
works as in how @kirby and @ilyakrasnov , I'm very unclear about what it needed here. I was asking for feedback on it several days ago, and I didn't get any feedback on it. I assumed passing tests means it works. Anything that should be done?

**Comment by Phatkid98** [comment]
I'll look into it an see what I can do though

**Comment by kirbygit** [comment]
@Phatkid98 if this is resolve please close the issue #31 and check if it works then ready to merge thx

**Comment by Phatkid98** [comment]
Hey @kirbygit ,it's not resolved. still working on making sure the available method works. I'll let you know for a review

**Pull request merged**